plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/building/N4JSBuildTypeTrackingBuilder.java

```diff
@@ -74,6 +74,12 @@ protected void doBuild(ToBeBuilt toBeBuilt, IProgressMonitor monitor, BuildType
 		try {
 			updateProjectReferencesIfNecessary();
 			N4JSBuildTypeTracker.setBuildType(getProject(), type);
+
```

Comment:
Please do not merge this. The index will be in a bogus state afterwards (not matching the files on disk), but persisted as if it was valid. This may lead to follow up surprises.